Move updateWrapper from IceGrowth

einola opened the pull request:
# Move updateWrapper from IceGrowth

Fixes #866 

### Task List
- [x] Defined the tests that specify a complete and functioning change (*It may help to create a [design specification & test specification](../../../wiki/Specification-Template)*)
- [x] Implemented the source code change that satisfies the tests
- [x] Documented the feature by providing worked example
- [x] Updated the README or other documentation
- [x] Completed the pre-Request checklist below

---
# Change Description

I replaced the overElements call to updateWrapper in IceGrowth with a call to iLateral->update(tst). The idea is to separate out the logic of lateral ice growth and melt out of IceGrowth itself and into ILateralIceSpread modules. This does require ILateralIceSpread.hpp to define and share three variables; for change in ice concentration, ice thickness, and snow thickness.

---
# Test Description

All CI tests pass.

---
# Documentation Impact

N/A

---
# Other Details

None

---
### Pre-Request Checklist

- [x] The requirements of this pull request are fully captured in an issue or design specification and are linked and summarised in the description of this PR
- [x] No new warnings are generated
- [x] The documentation has been updated (or an issue has been created to track the corresponding change)
- [x] Methods and Tests are commented such that they can be understood without having to obtain additional context
- [x] This PR/Issue is labelled as a bug/feature/enhancement/breaking change
- [x] File dates have been updated to reflect modification date
- [x] This change conforms to the conventions described in the README

einola committed:
Move updateWrapper from IceGrowth

I replaced the overElements call to updateWrapper in IceGrowth with a
call to iLateral->update(tst). The idea is to separate out the logic of
lateral ice growth and melt out of IceGrowth itself and into
ILateralIceSpread modules. This does require ILateralIceSpread.hpp to
define and share three variables; for change in ice concentration,
ice thickness, and snow thickness.

einola committed:
Initialise base class variables

It's necessary to pass the tests. Is it the right thing to do here ...
see the comment I added.

timspainNERSC commented on physics/src/modules/include/ILateralIceSpread.hpp:
```diff
@@ -20,53 +20,54 @@ class ILateralIceSpread : public ModelComponent {
+        newice.resize();
+        snowMelt.resize();
+
+        /* Set these to zero, so we don't have uninitialized values floating around.
+         * Should this be done here, or in derived classes? Arguably only in the derived classes,
+         * since having uninitialised values helps with error detection. But as long as we have the
+         * nextsim_thermo.use_thermo_forcing option, all derived classes must initialise these
+         * values to something sensible.*/
```
I'd limit this comment to what the code is doing and why. And leave out the uncertainty.
```
         /*
           * Set these to zero, so we don't have uninitialized values floating around.
           * As long as we have the nextsim_thermo.use_thermo_forcing option, all
           * derived classes must initialise these values to something sensible.
           */

einola replied:
Yes, good point.

timspainNERSC commented on physics/src/include/IceGrowth.hpp:
I believe the functions

- `newIceFormation`
- `lateralIceSpread`
- `applyLimits`
- `updateWrapper`

are now redundant and can be deleted from `IceGrowth.hpp`.

einola replied:
Good catch!

timspainNERSC reviewed (changes requested):
- A suggested change the the wording of a comment.
- Some more lines of code that I am fairly sure can be deleted.

einola committed:
Address Tim's GitHub comments

Removes unneded functions from IceGrowth.hpp and fixes a comment in
ILateralIceSpread.hpp.

einola committed:
I made a mistake editing the comment

All better now.
